### PR TITLE
Disable sourcemaps for production build

### PR DESCRIPTION
**Pull request by stefaniacardenas**
# Problem
Currently, `css-loader` when generating source maps, uses absolute paths in the `sources` array. These paths are visible in the Chrome developer tools which is not ideal.

Screen Shot 2019-08-20 at 17 39 35
Screen Shot 2019-08-20 at 17 40 13

# Solution
This seems to be a known bug in `css-loader`. Please refer to the following issues
https://github.com/webpack-contrib/css-loader/issues/635
https://github.com/webpack-contrib/css-loader/issues/652
https://github.com/webpack-contrib/css-loader/issues/886

There is a PR that fixes part of the issues related to source map paths but it does not seem to fix the problem of absolute paths inside the map `sources` array.
https://github.com/webpack-contrib/css-loader/pull/901

Therefore my solution involves disabling `sourceMap` for our NPM build.
I also bumped the version of `css-loader` and `style-loader`

## Checklist
_put `n/a` if item is not relevant to PR changes_

- [x] Have the CHANGELOG, README, MIGRATION, RELEASE_GUIDELINES docs been updated?
- [ ] Have new automated tests been implemented?
- [ ] Have new manual tests been written down?
- [x] Have tests passed locally?
- [ ] Have any new strings been translated?

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-771-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-771-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://771-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-771-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-771-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://771-pr-onfido-sdk-ui-onfido.surge.sh

File: webpack.config.babel.js [hunk 1]
```diff
@@ -62,38 +63,43 @@ const baseStyleLoaders = (modules=true) => [
...
-   use:
-    [
-      disableExtractToFile || !PRODUCTION_BUILD ?
-        'style-loader' :
-        MiniCssExtractPlugin.loader,
-      ...baseStyleLoaders(modules)
-    ]
- }))
+const baseStyleRules = (options={}) => {
+  const { disableExtractToFile, withSourceMap } = options
```
Comment (rfreitas) [inline webpack.config.babel.js]: why not deconstruct in the arguments declaration?

**Comment by rfreitas** [comment]
From reading https://github.com/webpack-contrib/css-loader/issues/652 it seems they fixed, but according to your test it didn't correct?

Also, found this https://github.com/webpack-contrib/less-loader/issues/80#issuecomment-287880842
> Starting with less-loader 4, source maps contain absolute paths now.
what?? 😨

File: webpack.config.babel.js [hunk 2]
```diff
@@ -62,38 +63,43 @@ const baseStyleLoaders = (modules=true) => [
...
+      modules: false
+    }].map(({rule, modules})=> ({
+      test: /\.(less|css)$/,
+      [rule]: [`${__dirname}/node_modules`],
+      use:
+       [
+         disableExtractToFile || !PRODUCTION_BUILD ?
+           'style-loader' :
+           MiniCssExtractPlugin.loader,
+         ...baseStyleLoaders(modules, withSourceMap)
```
Comment (rfreitas) [inline webpack.config.babel.js]: this will disable the sourcemap for both dist and npm

Comment (stefaniacardenas) [reply webpack.config.babel.js]: > From reading [webpack-contrib/css-loader#652](https://github.com/webpack-contrib/css-loader/issues/652) it seems they fixed, but according to your test it didn't correct?

Yes, I saw the source code with the fix but the problem was still there on upgrading. I couldn't figure out if I need to pass a parameter to the previous loader to make it work.
 
> Also, found this [webpack-contrib/less-loader#80 (comment)](https://github.com/webpack-contrib/less-loader/issues/80#issuecomment-287880842)
> 
> > Starting with less-loader 4, source maps contain absolute paths now.
> 
> what?? why?? 😨

Yes, I read that but from the most recent discussions it seems like they've changed their minds
https://github.com/webpack-contrib/css-loader/issues/886#issuecomment-456851963

Comment (stefaniacardenas) [reply webpack.config.babel.js]: > this will disable the sourcemap for both dist and npm
Oops, will have a look

Comment (kopijunkie) [reply webpack.config.babel.js]: I agree with @rfreitas on looking at the full file `baseStyleRules` is used in both the `configDist` and `configNpmLib`

Comment (stefaniacardenas) [reply webpack.config.babel.js]: Yes, I had a default value in place then I did some refactoring and I removed it by mistake

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-771-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-771-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://771-pr-onfido-sdk-ui-onfido.surge.sh

File: webpack.config.babel.js (hunk 2 above)
Comment (rfreitas) [reply webpack.config.babel.js]: it makes sense to me now 👍

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-771-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-771-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://771-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-771-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-771-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-771-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-771-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://771-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-771-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://771-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-771-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-771-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://771-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-771-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-771-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-771-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-771-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-771-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://771-pr-onfido-sdk-ui-onfido.surge.sh

**Pull request merged**